client/ayon_core/hosts/houdini/plugins/create/create_bgeo.py

```diff
@@ -61,45 +60,56 @@ def create(self, product_name, instance_data, pre_create_data):
 
         instance_node.setParms(parms)
 
+    @staticmethod
+    def update_node_parameters(node, creator_attributes):
+        """update node parameters according to creator attributes.
+
+        Implementation of update_node_parameters.
+
+        Args:
+            node(hou.Node): Houdini node to apply changes to.
+            creator_attributes(dict): Dictionary of creator attributes.
+        """
+
+        file_path, _ = lib.splitext(
+            node.evalParm("sopoutput"),
+            allowed_multidot_extensions=[
+                ".ass.gz", ".bgeo.sc", ".bgeo.gz",
+                ".bgeo.lzma", ".bgeo.bz2"
+            ]
+        )
+
+        output = "{file_path}.{ext}".format(
+            file_path=file_path,
+            ext=creator_attributes["bgeo_type"]
```

Comment:
You are calling this method with the `creator_attributs` from the `new_values` as such I suspect `bgeo_type` might NOT be available if its not changed and this will error?

Reply:
`update_instances` is only called when there's a change. and `new_values` include both the un-changed values as well as changed values.

Reply:
Are you sure? You are [here](https://github.com/ynput/ayon-core/pull/418/files/ae7097b6bdc43bb31a000272b6f964d910b1ceaf#diff-cd393404eac3ba072c67b8d5f1872e399e07a988c6e60d51227991ace4158090R254-R257) explicitly filtering to only those that have `changed_keys` which should really only be the 'new data' - not all the data. If it does contain all, that sounds like a bug, right @iLLiCiTiT ?

I'd expect `created_inst.data_to_store()` to return all data instead. E.g. like used [here](https://github.com/pypeclub/OpenPype/blob/dc27a9e608688657b726157b4ec633b33407506a/openpype/hosts/maya/api/plugin.py#L267-L272)

Reply:
here's a test from this PR. 
I printed the `creator_attributes` inside `update_node_parameters`
when creating a new redshift instance I got: 
```
{'farm': True, 'split_render': True, 'image_format': 'exr', 'multi_layered_mode': '1'}
```
and when I updated it from the publisher I got
```
{'multi_layered_mode': '2', 'farm': True, 'split_render': True, 'image_format': 'exr'}
```

Reply:
> Are you sure? You are [here](https://github.com/ynput/ayon-core/pull/418/files/ae7097b6bdc43bb31a000272b6f964d910b1ceaf#diff-cd393404eac3ba072c67b8d5f1872e399e07a988c6e60d51227991ace4158090R254-R257) explicitly filtering to only those that have changed_keys which should really only be the 'new data' - not all the data.

I checked it again and  this true! 
it only updates the `changed_keys` 
but, it works here well because the changed key is `creator_attributes`

```
changes.changed_keys:  {'creator_attributes'}
new_values:  {'creator_attributes': {'farm': True, 'split_render': True, 'image_format': 'exr', 'multi_layered_mode': '2'}}
```
So, the code will error if I changed any of these attributes for example.

![image](https://github.com/ynput/ayon-core/assets/20871534/ebac56f7-5670-4c38-bbb9-a80b829cb220)

Reply:
> but, it works here well because the changed key is `creator_attributes`

Yup, that's what I expected. I wanted to comment that before, but apparently I forgot. Glad you caught it.

Reply:
> You are calling this method with the creator_attributs from the new_values as such I suspect bgeo_type might NOT be available if its not changed and this will error?

since one key inside `creator_attributes` is changed then `creator_attributes` parameter is considered changed. and it's value is passed even the change is in one of its keys only. 

but, your concern was right as when changing `publish_attributes`, then  `creator_attributes`  won't be passed which leads to the error in my previous comment. 

That case should be solved in https://github.com/ynput/ayon-core/pull/418/commits/ad16bdf43168b8f8727f1bb15ab0d3ae90562602